### ISSUE TITLE
Add smoke tests for all domain modules

### ISSUE THREAD

**Issue by kiyeonjeon21** [kiyeonjeon21/latent-lab#6]
## Context

Currently there are only 2 test files covering utilities (8 assertions total). No domain modules are tested, so any change risks silent breakage.

## What needs to happen

Add a smoke test per domain that verifies `run(cfg)` executes without error on a minimal config. These don't need to validate ML accuracy — just that the code runs.

Example pattern:

```python
# tests/test_ml_smoke.py
from omegaconf import OmegaConf
from latent_lab.ml.classification import run

def test_ml_classification_smoke():
    cfg = OmegaConf.create({
        "name": "test_classification",
        "domain": "ml",
        "task": "classification",
        "data": {"name": "iris"},
        "model": {"name": "random_forest", "n_estimators": 10, "max_depth": 3},
        "training": {"seed": 42, "learning_rate": 0.1},
        "tags": ["test"],
    })
    run(cfg)  # should not raise
```

## Domains to cover

- [ ] `ml` — classification, regression (use small built-in datasets)
- [ ] `dl` — CNN with 1 epoch, tiny batch (may need to mock MPS if running in CI)
- [ ] `llm` — skip or mock (requires model download)
- [ ] `nlp` — skip or mock (requires model download)
- [ ] `cv` — skip or mock (requires model download)
- [ ] `rl` — classic CartPole with 1000 timesteps
- [ ] `rag` — skip or mock (requires ollama)

## Notes

- Tests that require model downloads or external services should be marked with `@pytest.mark.slow` or skipped in CI
- ML and RL smoke tests can run without any downloads — start there
- Run with: `uv run pytest tests/ -v`
